### PR TITLE
Feature/update object

### PR DESCRIPTION
**Pull request by Montana-Pfeifer**
Added #update to controller for merchants and items
route patch for merchants and put for items as seen on postman
100% testing

### CODE REVIEW

**Comment by TDManning** [comment]
@Montana-Pfeifer - In order to fix these merge conflicts, before we proceed, can you go to your local and do the following?

-git pull from origin main onto this working branch
-resolve conflict in  merchants_requsts_spec.rb file
-ignore coverage so that those other merge conflict files will not be included. Screen shot attached. @devklynch @Paul-Knapp @ossar3 should add this line /coverage to their gitignore as well. 
-commit changes and repush from this same branch. 
Screenshot 2024-10-30 at 3 28 30 PM

Thank you!

File: config/routes.rb [hunk 1]
```diff
@@ -13,10 +13,12 @@
...
   delete "/api/v1/merchants/:id", to: "api/v1/merchants#destroy"
   post "/api/v1/merchants", to: "api/v1/merchants#create"
+  patch "/api/v1/merchants/:id", to: "api/v1/merchants#update"
 
 #items below
   get "/api/v1/items", to: "api/v1/items#index"
   get "/api/v1/items/:id", to: "api/v1/items#show"
   delete "/api/v1/items/:id", to: "api/v1/items#destroy"
   post "/api/v1/items", to: "api/v1/items#create"
+  put "/api/v1/items/:id", to: "api/v1/items#update"
```
Comment (devklynch) [inline config/routes.rb:23]: We may have to ask Abdul/Heather if we should use put or Patch for this based on the requirements and postman

**Review by devklynch — APPROVED** [review]
Thanks for working through those conflicts and then I just made a comment to check that patch vs post verb choice

File: config/routes.rb (hunk 1 above)
Comment (TDManning) [reply config/routes.rb:23]: I recall this exact question coming up today and also think the instructors said to use PATCH based on the spec sheet.

File: Gemfile.lock [hunk 2]
```diff
@@ -229,7 +229,7 @@ GEM
       rubocop-ast (>= 1.32.2, < 2.0)
       ruby-progressbar (~> 1.7)
       unicode-display_width (>= 2.4.0, < 3.0)
-    rubocop-ast (1.32.3)
+    rubocop-ast (1.33.0)
```
Comment (TDManning) [inline Gemfile.lock:232]: Great! I was getting some errors with this on my local. Thank you!

**Pull request merged**